chain-time/src/timeframe.rs

```diff
@@ -0,0 +1,168 @@
+use crate::timeline::Timeline;
+use std::time::{Duration, SystemTime};
+
+/// Identify a slot in a *specific* timeframe
+///
+/// The slots are not comparable to others slots made on a
+/// different time frame
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct Slot(pub(crate) u64);
+
+impl From<Slot> for u64 {
+    fn from(s: Slot) -> u64 {
+        s.0
+    }
+}
+
+/// Identify a slot in a specific timeframe and a leftover duration
+#[derive(Debug)]
+pub struct SlotAndDuration {
```

Comment:
I would bikeshed this as a more concise `SlotTime`.

Reply:
I'm not a fan of either name, so leaving it be for now, until it get some service anyway

Reply:
Does this provide much added value over a `(Slot, Duration)` tuple?